src/components/TransactionHistoryRows/TransactionDetails/TransferColumn.tsx

```diff
@@ -1,19 +1,18 @@
 import { Stack, useColorModeValue } from '@chakra-ui/react'
-import type { TxTransfer } from '@shapeshiftoss/chain-adapters'
+import { fromAssetId } from '@shapeshiftoss/caip'
 import { AssetIcon } from 'components/AssetIcon'
-import { selectAssetById } from 'state/slices/assetsSlice/selectors'
-import { useAppSelector } from 'state/store'
+import type { Transfer } from 'hooks/useTxDetails/useTxDetails'
 
 import { Address } from './Address'
 import { Amount } from './Amount'
+import { NftId } from './NftId'
 import { Row } from './Row'
 
 type TransferColumnProps = {
   compactMode?: boolean
-} & TxTransfer
+} & Transfer
 
 export const TransferColumn = (transfer: TransferColumnProps) => {
-  const asset = useAppSelector(state => selectAssetById(state, transfer.assetId))
```

Comment:
not sure how heavy this can be for every transfer column, but we already have the assets on the transfers, so no need to refetch from store

Reply:
Just to bring a bit of context, there's not really refetch here but rather a reference invalidating - not a concern here since:

- `selectAssetById` is a simple O(1) access of `state.assets.byId` by `assetId` key
- we use a `createCachedSelector` re-reselect util so `selectAssetById()` calls will be cached for a given assetId, meaning the selector won't even re-run/invalidate

Still happy with this removed line and not using `useAppSelector` unless we have to though, that should be kept for when we do need the reactivity.
As mentioned above, selectors invalidate when their deps/output change (depending on the util we use to create them i.e `deepEqualOutput`, `cached` selector etc). If `asset` was to lose its reference (which won't be the case here) then the whole thing would re-render for no reason since we already have assets in transfer columns.

Reply:
Ok, thanks for the breakdown. It seems like we would get the same reactivity from `useTxDetails` if anything update that would then fall through to all sub components using transfers, so I was just seeing a lot of these extra useAppSelector calls that didn't seem necessary in this transfer context flow